init-tools.msbuild

```diff
@@ -1,14 +1,23 @@
-<Project Sdk="Microsoft.NET.Sdk">
+<Project>
+  <!-- BaseIntermediateOutputPath must be set before Microsoft.Common.props is loaded. --> 
+  <PropertyGroup>
+    <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)Tools/$(BuildToolsPackageVersion)/</BaseIntermediateOutputPath>
+  </PropertyGroup>
+
+  <Import Project="Directory.Build.props" />
+  <Import Project="Sdk.props" Sdk="Microsoft.NET.Sdk" />
+
   <PropertyGroup>
     <TargetFramework>netcoreapp1.0</TargetFramework>
     <EnableDefaultItems>false</EnableDefaultItems>
     <DisableImplicitFrameworkReferences>true</DisableImplicitFrameworkReferences>
-    <BaseIntermediateOutputPath>$(MSBuildThisFileDirectory)Tools/$(BuildToolsPackageVersion)</BaseIntermediateOutputPath>
```

Comment:
So this never worked before?

Reply:
Right, because it was set too late. The restore assets always landed in the intermediate path. But setting the semaphore (which is done in init-tools.cmd) worked of course because it was handled entirely in the batch script: `BUILD_TOOLS_SEMAPHORE_DIR=%TOOLRUNTIME_DIR%\%BUILDTOOLS_VERSION%`. 

With my change now the semaphore and the restore assets share the same directory.